tests/acceptance/stepDefinitions/sharingContext.js

```diff
@@ -160,14 +162,78 @@ const checkIsListedInAutoComplete = function (type, displayName) {
         displayNameWithType = displayName + '\nGroup'
       }
       return itemsList.includes(displayNameWithType)
-    })
+    }
+    )
+}
+
+const getUsersMatchingPattern = function (pattern) {
+  // check if all created users that contain the pattern either in the display name or the username
+  // are listed in the autocomplete list
+  // in both cases the display name should be listed
+  const users = userSettings.getCreatedUsers()
+  const usersID = Object.keys(users)
+  return usersID.filter(
+    id => users[id].displayname.toLowerCase().includes(pattern) || id.includes(pattern)
+  ).map(
+    id => users[id].displayname
+  )
+}
+
+const getGroupsMatchingPattern = function (pattern) {
+  const groups = userSettings.getCreatedGroups()
+  const groupMatchingPattern = groups.filter(group => group.toLowerCase().includes(pattern))
+  return groupMatchingPattern
+}
+
+const assertUsersInAutocompleteList = async function (usersMatchingPattern) {
+  const itemsList = await client.page.FilesPageElement.sharingDialog().getShareAutocompleteItemsList()
+  const userPostfix = client.page.FilesPageElement.sharingDialog().getUserSharePostfix()
+  const usersNotFound = _.difference(
+    usersMatchingPattern.map(name => name + userPostfix),
+    itemsList
+  )
+  assert.strictEqual(usersNotFound.length, 0, `could not find ${usersNotFound} in the itemsList`)
+}
+
+const assertGroupsInAutocompleteList = async function (groupMatchingPattern) {
+  const itemsList = await client.page.FilesPageElement.sharingDialog().getShareAutocompleteItemsList()
+  const groupPostfix = client.page.FilesPageElement.sharingDialog().getGroupSharePostfix()
+  const groupsNotFound = _.difference(
+    groupMatchingPattern.map(name => name + groupPostfix),
+    itemsList
+  )
+  assert.strictEqual(groupsNotFound.length, 0, `could not find ${groupsNotFound} in the itemsList`)
+}
+
+const getUsersWithThePattern = async function (pattern, alreadySharedUserOrGroup, userOrGroup) {
```

Comment:
Looks great. You just need to fix the name of this function.

```suggestion
const assertUsersGroupsWithPatternInAutocompleteListExcluding = async function (pattern, sharedUserOrGroup, userOrGroup) {
```

Reply:
If you have shorter name than this, please comment. :grin:

Reply:
Thankyou.I fixed that.